subsys/bluetooth/services/fast_pair/fp_registration_data.c

```diff
@@ -24,13 +25,16 @@ static const uint8_t fp_magic[] = {0xFA, 0x57, 0xFA, 0x57};
 #define FP_OFFSET(_prev_item_offset, _prev_item_len) \
 	(_prev_item_offset + ROUND_UP(_prev_item_len, FP_DATA_ALIGN))
 
-#define FP_MAGIC_HEADER_OFF	 FP_DATA_START
-#define FP_MODEL_ID_OFF		 FP_OFFSET(FP_MAGIC_HEADER_OFF, FP_MAGIC_SIZE)
+#define FP_MAGIC_OFF		 FP_DATA_START
+#define FP_MODEL_ID_OFF		 FP_OFFSET(FP_MAGIC_OFF, FP_MAGIC_SIZE)
 #define FP_ANTI_SPOOFING_KEY_OFF FP_OFFSET(FP_MODEL_ID_OFF, FP_REG_DATA_MODEL_ID_LEN)
-#define FP_MAGIC_TRAILER_OFF	 FP_OFFSET(FP_ANTI_SPOOFING_KEY_OFF, \
+#define FP_HASH_OFF		 FP_OFFSET(FP_ANTI_SPOOFING_KEY_OFF, \
 					   FP_REG_DATA_ANTI_SPOOFING_PRIV_KEY_LEN)
 
-BUILD_ASSERT(FP_OFFSET(FP_MAGIC_TRAILER_OFF, FP_MAGIC_SIZE) <= FP_PARTITION_SIZE,
+#define FP_DATA_SIZE		 (FP_OFFSET(FP_HASH_OFF, FP_CRYPTO_SHA256_HASH_LEN) - FP_DATA_START)
+#define FP_OFFSET_TO_DATA_IDX(_offset) (_offset - FP_DATA_START)
+
+BUILD_ASSERT(FP_OFFSET(FP_HASH_OFF, FP_CRYPTO_SHA256_HASH_LEN) <= FP_PARTITION_SIZE,
```

Comment:
Can we make it a more strict condition check and validate if the end offset is equal to the partition size?

Reply:
I think there is no need to do it. I would let user to create bigger partition if needed (e.g. user could allocate whole FLASH page to allow erasing it in order to update the data)